pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -42,6 +42,7 @@ var (
 	ErrFull              = errors.New("range is full")
 	ErrAllocated         = errors.New("provided IP is already allocated")
 	ErrMismatchedNetwork = errors.New("the provided network does not match the current range")
+	ErrMaskTooSmall      = errors.New("subnet mask must be longer than /1 for IPv4 or longer than /65 for IPv6")
```

Comment:
I'm not sure what /1 means here - do you mean /31 ?  `10.0.0.0/1` is nonsense?

Reply:
I mean the mask must be /2 or longer (/8, /16, etc.). I did not change any of the existing IPv4 logic in this PR. I updated the error message because it was incorrect.

__Old Error:__`masks greater than 31 bits are not supported`

As you can see in the RangeSize function, `(bits - ones) >= 31`. For IPv4, bits = 32 and ones is the number of masked bits in the 32-bit address. For a /1, the math would trigger the error ``(32 - 1) >= 31`.  I am happy to submit a PR to implement a fix if this is not the expected behavior, unless you prefer the fix to be implemented in this PR.